backend/migrations/20230310_000000_migration.sql

```diff
@@ -0,0 +1,329 @@
+-- DROP TABLE IF EXISTS toplevel_oplists CASCADE;
+-- DROP TABLE IF EXISTS traces_v0;
+-- DROP TABLE IF EXISTS user_data CASCADE;
+
+CREATE OR REPLACE FUNCTION trigger_set_timestamp()
+RETURNS TRIGGER AS $t$
+  BEGIN
+    NEW.updated_at = NOW();
+    RETURN NEW;
+  END;
+$t$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION canvas_id(_new_id uuid, _account_id uuid, _name VARCHAR(40), OUT _id uuid) AS
+  $func$
+  BEGIN
+  LOOP
+    SELECT id
+    FROM   canvases_v0
+    WHERE  name = _name
+    INTO   _id;
+
+    EXIT WHEN FOUND;
+
+    INSERT INTO canvases_v0 AS c
+    (id, account_id, name)
+    VALUES (_new_id, _account_id, _name)
+    ON     CONFLICT (name) DO NOTHING
+    RETURNING c.id
+    INTO   _id;
+
+    EXIT WHEN FOUND;
+  END LOOP;
+  END;
+$func$ LANGUAGE plpgsql;
+
+
+CREATE TABLE IF NOT EXISTS
+accounts_v0
+( id UUID PRIMARY KEY
+, username VARCHAR(255) UNIQUE NOT NULL
+);
+
+CREATE TRIGGER set_account_timestamp
+BEFORE UPDATE ON accounts_v0
+FOR EACH ROW
+EXECUTE PROCEDURE trigger_set_timestamp();
+
+
+CREATE TABLE IF NOT EXISTS
+canvases_v0
+( id UUID PRIMARY KEY
+, account_id UUID REFERENCES accounts_v0(id) NOT NULL
+, name VARCHAR(64) UNIQUE NOT NULL
+, created_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
+
+CREATE TRIGGER set_canvas_timestamp
+BEFORE UPDATE ON canvases_v0
+FOR EACH ROW
+EXECUTE PROCEDURE trigger_set_timestamp();
+
+
+CREATE TABLE IF NOT EXISTS
+cron_records_v0
+( id SERIAL PRIMARY KEY
+, tlid BIGINT NOT NULL
+, canvas_id UUID NOT NULL
+, ran_at TIMESTAMPTZ NOT NULL
+);
+
+CREATE INDEX IF NOT EXISTS
+idx_cron_records_tlid_canvas_id_id
+ON cron_records_v0
+(tlid, canvas_id, id DESC);
+
+
+CREATE TABLE IF NOT EXISTS
+custom_domains_v0
+( host TEXT PRIMARY KEY
+, canvas TEXT);
+
+
+CREATE TABLE IF NOT EXISTS
+events_v0
+( id UUID PRIMARY KEY
+, canvas_id UUID NOT NULL
+, module TEXT NOT NULL
+, name TEXT NOT NULL
+, modifier TEXT NOT NULL
+, locked_at TIMESTAMPTZ -- nullable
+, enqueued_at TIMESTAMPTZ NOT NULL
+, value TEXT NOT NULL
+);
+
+-- We want to use this index to:
+-- 1) count the number of items in this queue, so it's important that the entire
+-- search term is in the index or it will need to hit disk. This is true even though
+-- the module rarely changes
+-- 2) fetch the indexes for all items we're unpausing. This is rare so it's fine to
+
+CREATE INDEX IF NOT EXISTS
+idx_events_count
+ON events_v0 (canvas_id, module, name);
+
+
+CREATE TABLE IF NOT EXISTS
+function_arguments_v0
+( id UUID PRIMARY KEY
+, canvas_id UUID NOT NULL
+, tlid BIGINT NOT NULL
+, trace_id UUID NOT NULL
+, timestamp TIMESTAMPTZ NOT NULL
+, arguments_json TEXT NOT NULL
+);
+
+CREATE INDEX IF NOT EXISTS
+function_arguments_most_recent
+ON function_arguments_v0
+(canvas_id, tlid, timestamp DESC);
+
+CREATE INDEX IF NOT EXISTS
+function_arguments_for_trace
+ON function_arguments_v0
+(canvas_id, tlid, trace_id);
+
+
+CREATE TABLE IF NOT EXISTS
+function_results_v0
+( id BIGINT PRIMARY KEY
```

Comment:
Setting the id column as the primary key causes the following error `duplicate key value violates unique constraint` 
What is the intended function of the id?, and should we create a new primary key or is it possible to modify the existing id values to prevent duplicates?